lib/internal/encoding.js

```diff
@@ -337,15 +337,22 @@ class TextEncoder {
 
   encode(input = '') {
     validateEncoder(this);
-    return encodeUtf8String(`${input}`);
+    input = `${input}`;
+    if (input.length > 0) {
+      return encodeUtf8(input);
+    }
+    return new Uint8Array([]);
```

Comment:
Why not `new Uint8Array(0)ˋ ?

Reply:
I’m wondering if these fast paths are worth it. Yes, they speed things up in the case where the input is empty, but practically speaking, that is almost never going to be the case, so overall users probably pay an unnecessary cost for the zero-length check?

Reply:
I removed the empty-input fast path. Thanks for the review.